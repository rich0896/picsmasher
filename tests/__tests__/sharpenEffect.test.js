const { SharpenEffect } = require('../../src/effects.js');
const { createTestCanvas } = require('./helpers/testHelpers.js');

describe('SharpenEffect', () => {
    test('should apply sharpen effect correctly', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new SharpenEffect({ intensity: 1 });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith('Applying Sharpen Effect');
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });
});
