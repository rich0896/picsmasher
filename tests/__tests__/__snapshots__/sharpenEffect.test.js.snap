// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharpenEffect should apply sharpen effect correctly 1`] = `ImageData { width: 200, height: 200, data: [255, 255, 255, 255, 200, 255, 255, 255, 200, 255...], length: 160000 }`;
